Library/Homebrew/formula_installer.rb

```diff
@@ -625,9 +661,10 @@ def pour
       downloader.stage
     end
 
-    keg = Keg.new(formula.prefix)
-    keg.relocate_install_names Keg::PREFIX_PLACEHOLDER, HOMEBREW_PREFIX.to_s,
-      Keg::CELLAR_PLACEHOLDER, HOMEBREW_CELLAR.to_s, :keg_only => formula.keg_only?
+    if !pour_bottle? || formula.bottle.needs_relocation?
```

Comment:
I haven't fully reviewed this PR yet, so I may be wrong. But since I just wrote these lines early today, I have a few questions:

* Why check `pour_bottle?` inside `pour` method? This seems to be unnecessary. 
* From my understanding, here you want to stop relocating bottles which don't need to. But since we relocate all the text files for all bottles, it means all formulae need to relocate, at least for text files(e.g. `INSTALL_RECEIPT.json`)